src/Components/ArticleData/ArticleData.jsx

```diff
@@ -1,13 +1,42 @@
 import { useState, useEffect } from 'react';
 import { useParams } from 'react-router-dom';
-import { fetchArticleById } from '../../App';
+import { fetchArticleById, incPatchVotesByArticleId, decPatchVotesByArticleId } from '../../App';
 import "./ArticleData.css"
 import ArticleComments from '../ArticleComments/ArticleComments';
+import Button from '@mui/material/Button';
 
 const ArticleData = () => {
     const [article, setArticle] = useState({})
     const {article_id} = useParams()
     const [isLoading, setIsLoading] = useState(true)
+    const [isFetchingInc, setIsFetchingInc] = useState(false)
+    const [isFetchingDec, setIsFetchDec] = useState(false)
+
+    const incSendVotes = (article_id) => {
+        if (isFetchingInc) return;
+        setIsFetchingInc(true);
+        setArticle((currArticle) => {
+            console.log(currArticle);
+            if(currArticle.article_id === article_id ) {
+                return {...currArticle, votes: currArticle.votes + 1}
+            }
+            return currArticle
+        })
+        incPatchVotesByArticleId(article_id).then(() => setIsFetchingInc(false))
+    }
+
+    const decSendVotes = (article_id) => {
+        if(isFetchingDec) return;
+        setIsFetchDec(true)
+        setArticle((currArticle) => {
+            console.log(currArticle);
+            if(currArticle.article_id === article_id ) {
+                return {...currArticle, votes: currArticle.votes - 1}
```

Comment:
There's something about rerendering the entire thing that feels a little clunky Maybe it would be worth having 'votes' render separately and then you can hold votes in state like this:
1. initialise [votes, setVotes] = useState(0)
2. in useEffect, setVotes(article.votes) 
3. when upvoted etc, setVotes((curr) =>curr + 1 or cur + -1)